engine/resources/exec.go

```diff
@@ -800,3 +821,21 @@ func (obj *wrapWriter) Write(p []byte) (int, error) {
 func (obj *wrapWriter) String() string {
 	return obj.Buffer.String()
 }
+
+// checks that environment variable name is valid
+func isNameValid(varName string) error {
+	if varName == "" {
+		return fmt.Errorf("variable name cannot be an empty string")
+	}
+	i := 0
+	for i = 0; i < len(varName); i++ {
+		c := varName[i]
+		if i == 0 && '0' <= c && c <= '9' {
+			return fmt.Errorf("variable name cannot begin with number")
+		}
+		if !(c == '_' || '0' <= c && c <= '9' || 'a' <= c && c <= 'z' || 'A' <= c && c <= 'Z') {
```

Comment:
O.K., but double check there isn't a more succinct or idiomatic alternative in the `strings` package.

Reply:
Only in the strings package, or should I consider other packages too (like regex)

Reply:
Just look in strings for now. Regex should be used as a last resort.

Reply:
I can't find a more succinct way in the strings package.